modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -1429,18 +1429,19 @@ void redoIndexWithoutSideChains(Block block) {
     }
 
     /**
-     * Retrieve two generation blocks with unity protocol info with one lock.
+     * Retrieves two generation blocks with unity protocol info.
+     * <p>
+     * Always returns a 2-element array. If the blocks cannot be retrieved the array will contain null values.
+     * Block[0] is the parent block and has the given hash. Block[1] is the grandparent block.
      *
-     * @param hash given hash of the block
-     * @return the 2 generation block data have matched hash with unity protocol info. Block[0] is
-     *     the parent block, Block[1] is the grandParent block. The return might only contain the
-     *     parent block and still return the 2-elements array.
+     * @param hash the hash of the parent block
+     * @return the retrieved two generation blocks with unity protocol info
      */
     public final Block[] getTwoGenerationBlocksByHashWithInfo(byte[] hash) {
+        Block[] blockFamily = new Block[] { null, null};
         if (hash == null) {
-            return null;
+            return blockFamily;
```

Comment:
Will let AionBlockChainImpl.createNewMiningBlockInternal() throw NPE

Reply:
It would already throw and an NPE because the blockFamily is accessed without a null check.

Reply:
So We should just throw NPE if the input hash equal to null. instead of return values (or return null)

Same as the other method.

Reply:
Ok. I made the update.